test/unit/server_transaction_test.js

```diff
@@ -0,0 +1,489 @@
+        StellarSdk.Operation.payment({
+          destination:
+            'GASOCNHNNLYFNMDJYQ3XFMI7BYHIOCFW3GJEOWRPEGK2TDPGTG2E5EDW',
+          asset: StellarSdk.Asset.native(),
+          amount: '100.50'
+        })
+      )
+      .setTimeout(StellarSdk.TimeoutInfinite)
+      .build();
+    transaction.sign(keypair);
+
+    let blob = encodeURIComponent(
+      transaction
+        .toEnvelope()
+        .toXDR()
+        .toString('base64')
+    );
+    this.axiosMock
+      .expects('post')
+      .withArgs(
+        'https://horizon-live.stellar.org:1337/transactions',
+        `tx=${blob}`
+      )
+      .returns(Promise.resolve({ data: {} }));
+
+    this.server
+      .submitTransaction(transaction)
+      .then(function() {
+        done();
+      })
+      .catch(function(err) {
+        done(err);
+      });
+  });
+  it('adds metadata - tx was too small and was immediately deleted', function(done) {
+    const response = {
+      _links: {
+        transaction: {
+          href:
+            'https://horizon.stellar.org/transactions/db2c69a07be57eb5baefbfbb72b95c7c20d2c4d6f2a0e84e7c27dd0359055a2f'
+        }
+      },
+      hash: 'db2c69a07be57eb5baefbfbb72b95c7c20d2c4d6f2a0e84e7c27dd0359055a2f',
+      ledger: 22895637,
+      envelope_xdr:
+        'AAAAAIUAEW3jQt3+fbT6nCASA1/8RWdp9fJ2woxqPHZPQUH/AAAAZAEH/OgAAAAjAAAAAQAAAAAAAAAAAAAAAFyIDdQAAAAAAAAAAQAAAAAAAAADAAAAAAAAAAFCQVQAAAAAAEZK09vHmzOmEMoVWYtbbZcKv3ZOoo06ckzbhyDIFKfhAAAAAAAAAAEAAAACAAAAAwAAAAAAAAAAAAAAAAAAAAFPQUH/AAAAQOJlnAnmSv1igsU/LjpXvuCqS/EcnM7oxgyk4ElnCwOz9YUEcvhXuc9GS2Sz1fMxsWvV9dHhmu3HvBrsphVl5A8=',
+      result_xdr: 'AAAAAAAAAGQAAAAAAAAAAQAAAAAAAAADAAAAAAAAAAAAAAACAAAAAA==',
+      result_meta_xdr:
+        'AAAAAQAAAAIAAAADAV1cFQAAAAAAAAAAhQARbeNC3f59tPqcIBIDX/xFZ2n18nbCjGo8dk9BQf8AAAACU4RoUgEH/OgAAAAiAAAABQAAAAEAAAAAhD8BLsZFQEF33rKS6YopQUT3b6iLBG4nspe68/DBNBYAAAAAAAAAAAEAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBXVwVAAAAAAAAAACFABFt40Ld/n20+pwgEgNf/EVnafXydsKMajx2T0FB/wAAAAJThGhSAQf86AAAACMAAAAFAAAAAQAAAACEPwEuxkVAQXfespLpiilBRPdvqIsEbieyl7rz8ME0FgAAAAAAAAAAAQAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAA='
+    };
+
+    let keypair = StellarSdk.Keypair.random();
+    let account = new StellarSdk.Account(keypair.publicKey(), '56199647068161');
+    let transaction = new StellarSdk.TransactionBuilder(account, { fee: 100 })
+      // note that the values aren't actually the ones used in the response package!
+      // (I couldn't remember exactly what values I used to generate the transactions)
+      .addOperation(
+        StellarSdk.Operation.manageOffer({
+          selling: new StellarSdk.Asset(
+            'BAT',
+            'GBDEVU63Y6NTHJQQZIKVTC23NWLQVP3WJ2RI2OTSJTNYOIGICST6DUXR'
+          ),
+          buying: StellarSdk.Asset.native(),
+          amount: '0.0000001',
+          price: '0.0000001'
+        })
+      )
+      .setTimeout(StellarSdk.TimeoutInfinite)
+      .build();
+    transaction.sign(keypair);
```

Comment:
You could actually generate this transaction just once since you are using a mock. ~100 lines of code less.

Reply:
Good call